certora/applyMunging.patch

```diff
@@ -0,0 +1,125 @@
+     IMetaMorphoBase,
+     IMetaMorphoStaticTyping
+ } from "./interfaces/IMetaMorpho.sol";
+-import {Id, MarketParams, Market, IMorpho} from "../lib/morpho-blue/src/interfaces/IMorpho.sol";
++import {Id, MarketParams, Market, IMorphoHarness} from "../harness/IMorphoHarness.sol";
+ 
+ import {PendingUint192, PendingAddress, PendingLib} from "./libraries/PendingLib.sol";
+ import {ConstantsLib} from "./libraries/ConstantsLib.sol";
+ import {ErrorsLib} from "./libraries/ErrorsLib.sol";
+ import {EventsLib} from "./libraries/EventsLib.sol";
+-import {WAD} from "../lib/morpho-blue/src/libraries/MathLib.sol";
+-import {UtilsLib} from "../lib/morpho-blue/src/libraries/UtilsLib.sol";
+-import {SafeCast} from "../lib/openzeppelin-contracts/contracts/utils/math/SafeCast.sol";
+-import {SharesMathLib} from "../lib/morpho-blue/src/libraries/SharesMathLib.sol";
+-import {MorphoLib} from "../lib/morpho-blue/src/libraries/periphery/MorphoLib.sol";
+-import {MarketParamsLib} from "../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
+-import {IERC20Metadata} from "../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+-import {MorphoBalancesLib} from "../lib/morpho-blue/src/libraries/periphery/MorphoBalancesLib.sol";
+-
+-import {Multicall} from "../lib/openzeppelin-contracts/contracts/utils/Multicall.sol";
+-import {Ownable2Step, Ownable} from "../lib/openzeppelin-contracts/contracts/access/Ownable2Step.sol";
+-import {ERC20Permit} from "../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC20Permit.sol";
++import {WAD} from "../../lib/morpho-blue/src/libraries/MathLib.sol";
++import {UtilsLib} from "../../lib/morpho-blue/src/libraries/UtilsLib.sol";
++import {SafeCast} from "../../lib/openzeppelin-contracts/contracts/utils/math/SafeCast.sol";
++import {SharesMathLib} from "../../lib/morpho-blue/src/libraries/SharesMathLib.sol";
++import {MarketParamsLib} from "../../lib/morpho-blue/src/libraries/MarketParamsLib.sol";
++import {IERC20Metadata} from "../../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/IERC20Metadata.sol";
++
++import {Multicall} from "../../lib/openzeppelin-contracts/contracts/utils/Multicall.sol";
++import {Ownable2Step, Ownable} from "../../lib/openzeppelin-contracts/contracts/access/Ownable2Step.sol";
++import {ERC20Permit} from "../../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC20Permit.sol";
+ import {
+     IERC20,
+     IERC4626,
+@@ -34,7 +32,7 @@
+     ERC4626,
+     Math,
+     SafeERC20
+-} from "../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC4626.sol";
++} from "../../lib/openzeppelin-contracts/contracts/token/ERC20/extensions/ERC4626.sol";
+ 
+ /// @title MetaMorpho
+ /// @author Morpho Labs
+@@ -45,9 +43,7 @@
+     using UtilsLib for uint256;
+     using SafeCast for uint256;
+     using SafeERC20 for IERC20;
+-    using MorphoLib for IMorpho;
+     using SharesMathLib for uint256;
+-    using MorphoBalancesLib for IMorpho;
```

Comment:
why can you do that ?

Reply:
Because the interface IMorpho has been extended to IMorphoHarness, which includes all the functions which used to be defined in the periphery libraries of Morpho Blue. It's like saying that Morpho Blue is a big contract with all the functions of the periphery defined inside. This work better with the Certora prover because otherwise the libary functions get copied many times, and this lead to a huge contract that is difficult to verify

Reply:
indeed, i messed up between IMorphoHarness and IMetamorphoHarness